.github/workflows/ci.yml

```diff
@@ -0,0 +1,88 @@
+# Copyright 2020 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+# GitHub workflow reference:
+# https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions
+
+name: ci
+on: [push, pull_request]
+jobs:
+  smoke:
+    runs-on: ubuntu-latest
+    steps:
+      # Checks out syzkaller repo at the path.
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          path: gopath/src/github.com/google/syzkaller
+      # Caches everything in .cache dir, in partiuclar we want to cache go-build and golangci-lint stuff.
+      # For reference see:
+      # https://help.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows#using-the-cache-action
+      - name: cache
+        uses: actions/cache@v1
+        with:
+          path: .cache
+          key: cache
+      # Run make presubmit and convert all error messages to github format:
+      # https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-error-message
+      - name: run
+        shell: bash
+        run: |
+          export HOME="$PWD"
+          mkdir -p .cache
+          set -o pipefail
+          gopath/src/github.com/google/syzkaller/tools/syz-env make presubmit_smoke | \
+            sed -E 's#\s*(.+):([0-9]+):([0-9]+): (.+)#\0\n::error file=\1,line=\2,col=\3::\4#'
+      # Upload coverage report to codecov.io.
+      - name: codecov
+        uses: codecov/codecov-action@v1
+        with:
+          file: gopath/src/github.com/google/syzkaller/.coverage.txt
+  arch:
+    runs-on: ubuntu-latest
+    needs: [smoke]
+    if: ${{ needs.smoke.result == 'success' }}
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          path: gopath/src/github.com/google/syzkaller
+      - name: cache
+        uses: actions/cache@v1
+        with:
+          path: .cache
+          key: cache
+      - name: run
+        run: |
+          export HOME="$PWD"
+          mkdir -p .cache
```

Comment:
Is this missing a 'set -o pipefile' ?

Reply:
Oh, yes!
Perhaps we need to wrap this fragment into a script somewhere in tools.

Reply:
Done. I added a script. Let's see if it works on CI.